main/src/flight_sql/auth_middleware/basic_call_header_authenticator.rs

```diff
@@ -12,11 +16,15 @@ use crate::http::header::Header;
 #[derive(Clone)]
 pub struct BasicCallHeaderAuthenticator {
     instance: DBMSRef,
+    auth_cache: Arc<AuthCache<AuthCacheKey, User>>,
```

Comment:
Is it better to make auth_cache belong to dbms? so it can hidden in dbms.authenticate  interface function.